IDE/ROWLEY-CROSSWORKS-ARM/kinetis_hw.c

```diff
@@ -119,7 +121,34 @@ static void hw_rtc_init(void)
 {
     /* Enable RTC clock and oscillator */
     SIM->SCGC6 |= SIM_SCGC6_RTC_MASK;
-    RTC->CR |= RTC_CR_OSCE_MASK;
+
+    if (RTC->SR & RTC_SR_TIF_MASK) {
+        /* Resets the RTC registers except for the SWR bit */
+        RTC->CR |= RTC_CR_SWR_MASK;
+        RTC->CR &= ~RTC_CR_SWR_MASK;
+
+        /* Set TSR register to 0x1 to avoid the TIF bit being set in the SR register */
+        RTC->TSR = 1;
+    }
+
+    /* Disable RTC Interrupts */
+    RTC_IER = 0;
+
+    /* Enable OSC */
+    if ((RTC->CR & RTC_CR_OSCE_MASK) == 0) {
+        int i;
+
+        /* Turn on */
+        RTC->CR |= RTC_CR_OSCE_MASK;
+
+        /* Wait RTC startup delay */
+        for (i=0; i<10000; i++) {
```

Comment:
I'll document the actual RTC Oscillator startup delay requirement of 1000 us. Since this is bare-metal I don't have access to a normal delay_us type function so I picked a value. Thanks for the feedback!